### PR TITLE
Feature/display validator name

### PR DESCRIPTION
**Pull request by ERussel**
- move identity logic in a separate factory to reuse
- add display name to payouts list and details

### CODE REVIEW

File: fearless/Common/Services/PayoutRewardsService/PayoutRewardsService+Fetch.swift [hunk 1]
```diff
@@ -304,21 +304,49 @@ extension PayoutRewardsService {
         }
     }
 
+    func createIdentityFetchOperation(
+        dependingOn eraValidatorsOperation: BaseOperation<[EraIndex: [EraValidatorInfo]]>,
+        codingFactoryOperation _: BaseOperation<RuntimeCoderFactoryProtocol>
```
Comment (ef1rspb) [inline fearless/Common/Services/PayoutRewardsService/PayoutRewardsService+Fetch.swift:309]: `codingFactoryOperation` is unused

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+final class IdentityOperationFactory {
+    let requestFactory: StorageRequestFactoryProtocol
+
+    init(requestFactory: StorageRequestFactoryProtocol) {
+        self.requestFactory = requestFactory
+    }
+
+    private func createSuperIdentityOperation(
+        dependingOn coderFactoryOperation: BaseOperation<RuntimeCoderFactoryProtocol>,
+        accountIds: @escaping () throws -> [Data],
```
Comment (ef1rspb) [inline fearless/Modules/Staking/Operations/IdentityOperationFactory.swift:24]: [AccountId]

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        dependingOn superOperation: SuperIdentityOperation,
+        identityOperation: IdentityOperation,
+        chain: Chain
+    ) -> BaseOperation<[AccountAddress: AccountIdentity]> {
+        ClosureOperation<[AccountAddress: AccountIdentity]> {
+            let addressFactory = SS58AddressFactory()
+
+            let superIdentities = try superOperation.extractNoCancellableResultData()
+            let identities = try identityOperation.extractNoCancellableResultData()
+                .reduce(into: [String: Identity]()) { result, item in
```
Comment (ef1rspb) [inline fearless/Modules/Staking/Operations/IdentityOperationFactory.swift:53]: suggest `[AccountAddress: Identity]`

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+                        identity: identity.info
+                    )
+                }
+            }
+        }
+    }
+
+    private func createIdentityWrapper(
+        dependingOn superIdentity: SuperIdentityOperation,
+        runtime: BaseOperation<RuntimeCoderFactoryProtocol>,
```
Comment (ef1rspb) [inline fearless/Modules/Staking/Operations/IdentityOperationFactory.swift:103]: runtimeOperation?

**Review by ef1rspb — APPROVED** [review]

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift (hunk 4 above)
Comment (ERussel) [reply fearless/Modules/Staking/Operations/IdentityOperationFactory.swift]: fixed

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift (hunk 3 above)
Comment (ERussel) [reply fearless/Modules/Staking/Operations/IdentityOperationFactory.swift]: fixed

File: fearless/Modules/Staking/Operations/IdentityOperationFactory.swift (hunk 2 above)
Comment (ERussel) [reply fearless/Modules/Staking/Operations/IdentityOperationFactory.swift]: fixed

File: fearless/Common/Services/PayoutRewardsService/PayoutRewardsService+Fetch.swift (hunk 1 above)
Comment (ERussel) [reply fearless/Common/Services/PayoutRewardsService/PayoutRewardsService+Fetch.swift]: fixed

**Pull request merged**